tools/preprocess_lancet_intervals.cwl

```diff
@@ -40,22 +44,23 @@ arguments:
               cmd += "vcf2bed --snvs < " + inputs.mutect2_vcf.basename +  " | cut -f 1-3 > mutect2.snvs.bed;";
               bed.push("mutect2.snvs.bed");
           }
-        if(flag == 0){
-            cmd += "echo \"No input vcfs found to convert.  Returning ref bed\"; cp " + inputs.ref_bed.path + " " + inputs.output_basename + ".lancet_intvervals.bed;";
-        }
-        else{
-            cmd += "cat " + bed.join(" ") + " " + inputs.ref_bed.path + " | bedtools sort | bedtools merge > " + inputs.output_basename + ".lancet_intvervals.bed;";
+          if(flag == 0){
+              cmd += "echo \"No input vcfs found to convert.  Returning ref bed\"; cp " + inputs.ref_bed.path + " " + inputs.output_basename + ".lancet_intvervals.bed;";
+          }
+          else{
+              cmd += "cat " + bed.join(" ") + " " + inputs.ref_bed.path + " | bedtools sort | bedtools merge > " + inputs.output_basename + ".lancet_intvervals.bed;";
+          }
+          return cmd;
         }
-        return cmd;
       }
 
 inputs:
   strelka2_vcf: {type: ['null', File], doc: "PASS vcf from strelka2 run for the sample to be analyzed"}
   mutect2_vcf: {type: ['null', File], doc: "PASS vcf from mutect2 run for the sample to be analyzed"}
-  ref_bed: {type: File, doc: "Exonic bed file recommended.  Will be augmented with strelka2 and/or mutect2 sites if supplied"}
+  ref_bed: {type: 'File?', doc: "Optional exonic bed file recommended.  Will be augmented with strelka2 and/or mutect2 sites if supplied"}
   output_basename: string
 outputs:
   run_bed:
-    type: File
+    type: 'File?'
```

Comment:
I don't think the lancet subwf will run without this output

Reply:
It already does in WXS. It just uses the original gatk_intervallisttools output (https://github.com/kids-first/kf-somatic-workflow/blob/master/workflow/kfdrc_production_WES_somatic_variant_cnv_wf.cwl#L199) instead of doing the preprocess -> gatk_intervallisttools (https://github.com/kids-first/kf-somatic-workflow/blob/master/workflow/kfdrc_production_WGS_somatic_variant_cnv.cwl#L277-L297) stuff.